### ISSUE TITLE
Unexpected errors with future library imports

### ISSUE THREAD

**Issue by jscn** [PyCQA/pylint#1160]
Thanks for such a great library!

I've run into an issue after upgrading from v1.5.5 to v1.6.4 where I'm getting a lot of `wrong-import-position` errors for imports which occur after importing the and installing the `standard_library` from the `future` library (as described [here](http://python-future.org/imports.html#aliased-imports), this needs to happen *before* imports of the standard library code).

### Steps to reproduce
1. Create a file like [this](https://gist.github.com/jscn/8eb050aed40abb9953ce0ac5eb91a873)
2. Run `pylint` on the file

### Current behavior
Using no config file at all, `pylint` reports the following errors:
```
C:  9, 0: Import "from collections import namedtuple" should be placed at the top of the module (wrong-import-position)
C:  9, 0: standard import "from collections import namedtuple" comes before "from future import standard_library" (wrong-import-order)

```

### Expected behavior
I'd love for `pylint` to just ignore the `future` imports when it's doing the import order/position checks (i.e., it should still report whether non-`future` library imports occur in the correct order *relative to each other*, but should totally ignore the `future` library imports themselves). But I understand it may be out of scope for `pylint` to have special knowledge of `future`.

If that's the case, then I'd like to be able to tell pylint to ignore a block of import statements entirely. It may already be possible to do this but I couldn't see how from the documentation. My only option as I understand it is to turn off the `wrong-import-position` check for the entire block of import statements. But I *want* pylint to tell me if my non-`future` related imports are ordered incorrectly!

So something like:
```python
# pylint: disable=something-or-other
from future import standard_library
standard_library.install_aliases()
# pylint: enable=something-or-other

from datetime import datetime
from collections import namedtuple
```
should still report `wrong-import-order` for the the `datetime` and `collections` imports, but should not report any `wrong-import-position` errors.

### pylint --version output
```
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 2.7.9 (default, Jun 29 2016, 13:08:31) 
[GCC 4.9.2]
```

**Comment by jscn** [comment]
Looks like whatever is causing this was introduced in v1.6.0

**Comment by PCManticore** [comment]
Hi @jscn 

Thank you for reporting this issue. Indeed, the request does seem reasonable. What would be easier to implement, though, would be an option which specifies the imports which should be omitted when computing the expected order, rather than taking care of the ignored blocks.

I am thinking that something like ``--ignored-imports-from-ordering=future`` could work.

**Comment by The-Compiler** [comment]
I guess that change would need to be done in https://github.com/timothycrosley/isort - right?

Also, wouldn't `--known-standard-library=future` suppress this already?

**Comment by PCManticore** [comment]
Unfortunately no, we still emit wrong-import-position, because we are seeing ``standard_library.install_aliases()`` as a separate operation that is splitting the imports in two groups. Without it, ``--known-standard-library`` indeed helps.

**Comment by The-Compiler** [comment]
But how would `--ignored-imports-from-ordering=future` help then? Wouldn't we need something like a `# pylint: this-needs-to-be-here` annotation for `standard_library.install_aliases() ` then? :wink:

**Comment by PCManticore** [comment]
It will simply tell pylint not to take in consideration the import nodes with that qualified name. I would not go to a ``pylint: this-needs-to-be-here`` feature, since this will:

  * add a new pylint-specific pragma in source code. Having just the enable / disable pragmas should be enough for now, an option being more portable.
  * require coordination with the file parsing module, which I am not keen to do.

**Comment by solbrigm** [comment]
I am also hitting this issue.

**Comment by jdanekrh** [comment]
As for a workaround, I had some luck following advice in https://stackoverflow.com/a/36829884/1047788

*Step 1* Replace the problematic

```
from future import standard_library
standard_library.install_aliases()
```

with

```
# noinspection PyUnresolvedReferences
import install_aliases  # noqa
```

*Step 2* Create `install_aliases.py` in root of your project, containing

```
from future import standard_library
standard_library.install_aliases()
```

*Step 3* In your `pylint.rc`, set

```
[IMPORTS]

known-standard-library=
    install_aliases
```

Only remaining problem with this is that Pycharm wants to reorder the `import install_aliases` line. I haven't yet found out how to tell it that this is 'standard library' as well.

**Comment by DanielNoord** [comment]
This is related to Python 2 which we no longer support so I'm closing this. Any other issues with `wrong-import-position` should probably have their own issue.
